ExampleMod/Content/TileEntities/BasicTileEntity.cs

```diff
@@ -0,0 +1,239 @@
+				NetMessage.SendTileSquare(Main.myPlayer, topLeft.X, topLeft.Y, tileData.Width, tileData.Height);
+				NetMessage.SendData(MessageID.TileEntityPlacement, number: topLeft.X, number2: topLeft.Y, number3: Type);
+				return -1;
+			}
+
+			return Place(topLeft.X, topLeft.Y);
+		}
+	}
+
+	// BasicTileEntityTile is the Tile that BasicTileEntityTile attaches to.
+	// The most important parts are the TileObjectData.newTile.HookPostPlaceMyPlayer assignment and the KillMultiTile method.
+	// All of the other methods show very commonly desired functionality, but are not required for a working Tile Entity.
+	public class BasicTileEntityTile : ModTile
+	{
+		public static LocalizedText StatusText { get; private set; }
+
+		public override void SetStaticDefaults() {
+			// Properties
+			Main.tileFrameImportant[Type] = true;
+			Main.tileNoAttach[Type] = true;
+			Main.tileLavaDeath[Type] = true;
+			TileID.Sets.DisableSmartCursor[Type] = true;
+
+			// Placement
+			TileObjectData.newTile.CopyFrom(TileObjectData.Style2x2);
+
+			// Tell the tile to place the Tile Entity on the tile after placing it.
+			TileObjectData.newTile.HookPostPlaceMyPlayer = new PlacementHook(ModContent.GetInstance<BasicTileEntity>().Hook_AfterPlacement, -1, 0, true);
+
+			// The additional "states" in BasicTileEntityTile.png are laid out vertically. If additional styles were added to this example later we'd want those placed horizontally.
+			TileObjectData.newTile.StyleHorizontal = true;
+
+			TileObjectData.addTile(Type);
+
+			// Etc
+			AddMapEntry(new Color(200, 200, 200), CreateMapEntryName(), MapHoverText);
+
+			StatusText = this.GetLocalization(nameof(StatusText));
+		}
+
+		public override void KillMultiTile(int i, int j, int frameX, int frameY) {
+			// When the tile is removed, we need to remove the Tile Entity as well.
+			ModContent.GetInstance<BasicTileEntity>().Kill(i, j);
+		}
+
+		// The following hooks all show accessing the Tile Entity and using it to adjust the behavior and look of this Tile.
+
+		public static string MapHoverText(string name, int i, int j) {
+			if (TryGetBasicTileEntity(i, j, out BasicTileEntity tileEntity)) {
+				return StatusText.Format(tileEntity.WaterFillPercentage);
+			}
+
+			// Note that it is possible for a map entry to be queried for a tile location that doesn't have a TileEntity anymore.
+			// This can happen in multiplayer when a world section hasn't been synced yet or when a players map hasn't been updated to match changes to the world.
+			// This code shows detecting those situations, but the basic lesson is don't assume a TileEntity will always be present.
+			/*
+			Point16 topLeft = TileObjectData.TopLeft(i, j);
+			int index = ModContent.GetInstance<BasicTileEntity>().Find(topLeft.X, topLeft.Y);
+			if (index == -1) {
+				if (!Main.sectionManager.TileLoaded(topLeft.X, topLeft.Y)) {
+					return $"{name}: World section not loaded yet";
+				}
+				return $"{name}: No TileEntity found at coordinate";
+			}
```

Comment:
Is there a reason you wouldn't chain this into the `if` above? No need to requery the `Find` right? `MapHoverText` is only called for our tiles, and `TryGetBasicTileEntity` can only fail if the section isn't loaded, or the TE is broken